x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/OAuth2TokenAuthenticator.java

```diff
@@ -62,4 +88,11 @@ public void authenticate(Context context, ActionListener<AuthenticationResult<Au
             listener.onFailure(e);
         }));
     }
+
+    private Map<String, Object> buildMetricAttributes(BearerToken token, String failureReason) {
+        if (failureReason != null) {
+            return Map.of(ATTRIBUTE_AUTHC_FAILURE_REASON, failureReason);
+        }
+        return Map.of();
```

Comment:
There is no really any attribute that we can attach to the collected token metrics that isn't considered sensitive. Hence we only attach failure reason if provided.